idunn/blocks/opening_hour.py

```diff
@@ -0,0 +1,77 @@
+import re
+import logging
+import datetime
+from apistar import types, validators
+import humanized_opening_hours as hoh
+from humanized_opening_hours.exceptions import (ParseError, SpanOverMidnight)
+
+def from_hours(hbnt):
+    """
+    Takes the "hours before next change" string as input and returns an int of the number of seconds.
+
+    >>> from_hours("")
+    0
+    >>> from_hours("5 minutes")
+    300
+    >>> from_hours("5 hours and 35 minutes")
+    20100
+    """
+    seconds = 0
+
+    hours = re.search(r"(\w+) hours", hbnt)
+    if hours:
+        seconds += int(hours.group(1)) * 3600
+
+    minutes = re.search(r"(\w+) minutes", hbnt)
+    if minutes:
+        seconds += int(minutes.group(1)) * 60
+    return seconds
+
+class OpeningHourBlock(types.Type):
+    type = validators.String(title="type")
+    status = validators.String()
+    next_transition_time = validators.String()
+    seconds_before_next_transition = validators.Integer()
+    is_24_7 = validators.Boolean()
+    raw = validators.String()
+    days = validators.Array()
+
+    @classmethod
+    def from_es(cls, es_poi, lang):
+        logger = logging.getLogger('idunn')
+        is247 = False
+
+        raw = es_poi.get('properties', {}).get('opening_hours')
+        if raw is None:
+            return None
+        elif raw is '24/7':
+            is247 = True
+
+        try:
+            clean_raw = hoh.OHParser.sanitize(raw)
+            oh = hoh.OHParser(clean_raw)
+        except (SpanOverMidnight, ParseError) as err:
+            logger.error(f"OpeningHour error: {err}")
+            return None
+
+        status = 'close'
+        if oh.is_open():
+            status = 'open'
+
+        ntt = oh.next_change().strftime('%H:%M')
+        ohr = hoh.render_field(clean_raw, universal=False)
+        sbnt = from_hours(ohr.time_before_next_change(word=False))
+
+        return cls(
+            type='OpeningHourBlock',
+            status=status,
+            next_transition_time=ntt,
+            seconds_before_next_transition=sbnt,
+            is_24_7=is247,
+            raw=clean_raw,
+            days=[] # format definition left for future work
+        )
+
+if __name__ == "__main__":
```

Comment:
hum why do you need this ?
we should use pytest to run the doc test, not mains

Reply:
done